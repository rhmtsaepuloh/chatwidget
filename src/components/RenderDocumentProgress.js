import React from "react";
import { FaRegTimesCircle } from "react-icons/fa";
import { MdSchedule } from "react-icons/md";

const RenderDocumentProgress = ({
  icon,
  text,
  progressUpload,
  timestamp,
  cancelUpload,
  caption,
}) => {
  return (
    <React.Fragment>
      <div className='file-document flex items-center justify-between bg-blue-200 w-full rounded px-1 py-3'>
        <div className='flex items-center justify-start w-1/2'>
          {icon}
          <p className='whitespace-pre-wrap break-words ml-2 w-3/4 text-xs'>
            {text}
          </p>
        </div>
        <div className='flex items-center justify-end'>
          <p className='text-blue-600 text-sm font-bold mr-1'>
            {progressUpload}%
          </p>
          <FaRegTimesCircle
            className='cursor-pointer'
            onClick={() => cancelUpload()}
          />
        </div>
      </div>
      <p className='whitespace-pre-wrap break-words mb-3'>{caption}</p>
      <div className='file-timestamp flex items-center justify-end'>
        <p className='text-sm' style={{ color: "#979797" }}>
          {timestamp}
        </p>
        <MdSchedule className='ml-1' />
      </div>
    </React.Fragment>
  );
};

export default RenderDocumentProgress;
