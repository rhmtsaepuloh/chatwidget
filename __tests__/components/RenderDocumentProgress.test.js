/* eslint-disable react/react-in-jsx-scope */
import React from "react";
import { shallow } from "enzyme";
import { FaFilePdf } from "react-icons/fa";
import RenderDocumentProgress from "../../src/components/RenderDocumentProgress";

const props = {
  icon: <FaFilePdf />,
  text: "text",
  timestamp: "10:20",
  cancelUpload: jest.fn(),
  downloadFile: jest.fn(),
  caption: "test",
};

describe("render correctly render document progress components", () => {
  test("renders if progress is 100", () => {
    const wrapper = shallow(
      <RenderDocumentProgress {...props} progressUpload={100} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test("renders if progress > 0", () => {
    const wrapper = shallow(
      <RenderDocumentProgress {...props} progressUpload={80} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
