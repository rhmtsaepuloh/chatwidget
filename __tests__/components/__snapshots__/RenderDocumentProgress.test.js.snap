// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly render document progress components renders if progress > 0 1`] = `ShallowWrapper {}`;

exports[`render correctly render document progress components renders if progress is 100 1`] = `ShallowWrapper {}`;
